Automattic/jetpack
# Writing Prompt block: better error handling when fetching prompt

## Proposed changes:

- Fixes an issue where REST API error responses were not passed through correctly
- Adds more specific error message when a prompt is not found (because of an invalid id)
- Adds a retry or reset action after the error message

| **Prompt not found** | **Generic error** |
| - | - |
| <img width="732" alt="image" src="https://user-images.githubusercontent.com/1699996/223570166-005c2970-08ae-4637-8c24-a8dcf695bc28.png"> | <img width="714" alt="image" src="https://user-images.githubusercontent.com/1699996/223570185-58b420fc-b763-4c95-be3c-660fc95cc42d.png"> |

### Other information:


## Jetpack product discussion

pe7F0s-uo-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

- Set up your wpcom sandbox for testing by applying the patch D102925-code and applying Jepack trunk to your sandbox with `bin/jetpack-downloader test jetpack trunk`
- Set the JETPACK__SANDBOX_DOMAIN constant to your wpcom sandbox host, using a define statement on your development site, or Settings > Jetpack Constants on your jurassic.ninja site.
- Make sure your site is connected to WP.com (you will need to use local tunneling if using a local dev site PCYsg-GJ2-p2)
- Insert a writing prompt block
- Use the code editor option in Gutenberg to change the promptId of the block to be invalid (e.g. `"promptId":9999`)
- Go back to the visual editor and you should see an error with the prompt id you set
- Click "Reset prompt" and the prompt should reload with the prompt for today
- To trigger the generic error, you can remove the JETPACK__SANDBOX_DOMAIN constant and unsandbox the API (before it's released to wpcom), which should trigger an error
- Click "Retry" to try and fetch the prompt again
